lib/email/sender.rb

```diff
@@ -514,5 +474,144 @@ def self.bounceable_reply_address?
     def self.bounce_address(bounce_key)
       SiteSetting.reply_by_email_address.sub("%{reply_key}", "verp-#{bounce_key}")
     end
+
+    def add_identification_field_headers(topic, post)
```

Comment:
I made come comments below, but on getting to `add_experimental_identification_field_headers` I think they might all be misguided. Reading `add_experimental_identification_field_headers` now...